src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -266,8 +266,8 @@ bool CanTokenFollowTypeInPattern(Precedence precedence)
                 case SyntaxKind.CommaToken:
                 case SyntaxKind.SemicolonToken:
                     return true;
-                case SyntaxKind.DotToken:
-                    // int.MaxValue is an expression, not a type.
+                case SyntaxKind.MinusGreaterThanToken: // parse as an expression for error recovery
```

Comment:
Is there a helper to check for "continuation of an expression"?

Reply:
Sorry. I didn't follow the question.

Reply:
Currently DotToken is added as a special case. I added MinusGreaterThanToken as another. I was looking for a proper helper to catch any such token that tells us that this should be parsed as an expression instead of a type.

Reply:
Ok I think MinusGreaterThanToken addition is fine, more kinds can be added when discovered.
Using a helper means we should list another set of tokens - that doesn't seem like an improvement compared to current situation.